File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
import math
from itertools import permutations

def isPrime(n):
    if n <= 1: return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0: return False
    return True

def solution(numbers):
    answer = set()
    nums = list(numbers)
    
    def get_permutation(arr, fix):
        for i in range(len(arr)):
            copy = arr[:i] + arr[i+1:]
            newN = int(fix + arr[i])
            if newN not in answer and isPrime(newN): answer.add(newN)
            get_permutation(copy, fix + arr[i])
    
    get_permutation(nums, '')
    return len(answer)

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function isPrime(num) {
    if (num < 2) return false;
    if (num === 2) return true;
    for (var i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) return false;
    }
    return true;
}

function solution(numbers) {
    let answer = 0;

    let n = numbers.split('');
    let nums = new Set()

    function combi(a, s) {
        if (s.length > 0) {
            if (nums.has(+s) === false) {
                nums.add(+s);
                if (isPrime(+s)) answer++;
            }
        }
        if (a.length > 0) {
            for (let i = 0; i< a.length; i++) {
                let t = a.slice(0)
                t.splice(i,1);
                combi(t,s + a[i]);
            }
        }
    }
    
    combi(n,'');

    return answer;
}